sdk/core/Azure.Core/src/Pipeline/Internal/ResponseBodyPolicy.cs

```diff
@@ -75,7 +75,9 @@ private async ValueTask ProcessAsync(HttpMessage message, ReadOnlyMemory<HttpPip
 
             if (message.BufferResponse)
             {
-                if (networkTimeout != Timeout.InfiniteTimeSpan)
+                // If cancellation is possible (whether due to network timeout or a user cancellation token being passed), then
+                // register callback to dispose the stream on cancellation.
+                if (networkTimeout != Timeout.InfiniteTimeSpan || oldToken.CanBeCanceled)
```

Comment:
Should we do this to cover any tokens linked to this source?
```suggestion
                if (networkTimeout != Timeout.InfiniteTimeSpan || cts.Token.CanBeCanceled)
```

Reply:
Nope, that one will always return true because it is a new linked token that we created.